files/templates/scrapeconfigs/additional-scrape-configs.template.yaml

```diff
@@ -527,36 +548,30 @@
   - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring, __meta_kubernetes_service_labelpresent_giantswarm_io_monitoring]
     regex: .*(true).*
     action: keep
-  - source_labels: [__meta_kubernetes_service_labelpresent_operated_prometheus]
-    regex: (true)
-    action: drop
-  - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring_port]
-    regex: (true)
-    action: keep
-  - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring_path]
-    regex: (true)
-    action: keep
-  - regex: (.*)
-    target_label: __address__
-    replacement: [[ .APIServerURL ]]:443
+    # if __meta_kubernetes_service_annotation_giantswarm_io_monitoring_path is present, we use it as the metrics path
+  - source_labels: [__meta_kubernetes_service_annotation_giantswarm_io_monitoring_path]
+    action: replace
+    target_label: __metrics_path__
+    regex: (.+)
+    # if __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port, we use it as the metrics port
+  - source_labels: [__address__, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port]
     action: replace
-  - source_labels: [__meta_kubernetes_namespace, __meta_kubernetes_pod_name, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_path]
-    regex: (.*);(.*);(.*);(.*)
+    target_label: __address__
+    regex: ([^:]+)(?::\d+)?;(\d+)
```

Comment:
This should work too, why bother if we do not need the second port ? 
`([^:]+).*;(\d+)`

But I'm not strong on this.

Reply:
I took this from how prometheus operator should be scraping: https://gist.github.com/reachlin/a98b90afcbff4604c90c183a0169474f#file-prometheus-yml-L33  
Basically, the following means:
([^:]+) = address
(?::\d+)? = port if it exists. This specific regex just ensures we do not capture the port value so we can do $1:$2, otherwise, we would need to do $1:$3